capstone-rs/src/arch/x86.rs

```diff
@@ -218,8 +223,23 @@ impl Default for X86Operand {
 impl<'a> From<&'a cs_x86_op> for X86Operand {
     fn from(op: &cs_x86_op) -> X86Operand {
         let op_type = X86OperandType::new(op.type_, op.__bindgen_anon_1);
+
```

Comment:
To eliminate the possibility for undefined behavior, do a separate bitwise check for read then write. Then check for cases ReadWrite, Read, and Write. In the else block, return Invalid.

Otherwise, other higher order bits being set could result in undefined behavior.

Reply:
Perhaps bitwise check is redundant. We only need to update it if capstone itself gets updated.

Currently, [`cs_ac_type`](https://github.com/aquynh/capstone/blob/fa17ed2453bd87f4aec72842571aed893c3a1963/include/capstone/capstone.h#L203) only has read and write operation defined. As the documentation clearly states that the operations are platform-independent, it's very unlikely that new bits will be added in the future.

Suppose that future-proof is *really* needed, the proposed API (`enum {R, W, RW}`) should be abandoned as a side-effect: we can't enumerate all combinations. Exporting `cs_ac_type` as a bitflag is future proof, but the downside is obvious: we are delegating the responsibility of decoding cs_ac_type to the user. This design spoils the consistent style of high-level API which `capstone-rs` exposes.